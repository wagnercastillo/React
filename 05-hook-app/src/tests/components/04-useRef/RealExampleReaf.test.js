import React from "react";
import { shallow } from "enzyme";
import { RealExampleReaf } from "../../../components/04-useRef/RealExampleReaf";
import { MultipleCustomHooks } from "../../../components/03-examples/MultipleCustomHooks";

describe("Pruebas en <RealExampleReaf />", () => {
	const wrapper = shallow(<RealExampleReaf />);

	test("debe de mostrarse correctamente", () => {
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
	});

	test("debe de mostrar el componente <MultipleCustomHooks />", () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
	});
});
