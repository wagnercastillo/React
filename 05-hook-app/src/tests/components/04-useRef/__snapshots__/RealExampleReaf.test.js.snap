// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleReaf /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Real Example Reaf
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
